x-pack/plugins/lens/public/editor_frame_service/editor_frame/suggestion_helpers.ts

```diff
@@ -160,6 +171,15 @@ export function getSuggestions({
     .sort((a, b) => b.score - a.score);
 }
 
+interface StateWithLayers {
+  [prop: string]: unknown;
+  layers: Array<Record<string, unknown>>;
+}
+interface SuggestionMultipleLayers extends Suggestion {
```

Comment:
This seems a bit hacky to me - we are assuming a certain shape of the internal state of visualizations and datasources in the frame which is meant to be generic. Further down below it's even mentioning yRightExtent, yLeftExtent, gridlines and so on.

Let's move all the state manipulation logic (stuff that doesn't treat the datasource state / visualization state as `unknown`) into the datasource / the xy visualization and introduce a new function like the new `updateLayersConfigurationFromContext` for it.